packages/compass-e2e-tests/helpers/commands/set-connect-form-state.ts

```diff
@@ -358,25 +358,32 @@ export async function setConnectFormState(
       );
 
       let found = false;
-      const options = await browser.$$('#select-key-menu [role="option"]');
-      const allText = [];
-      for (const option of options) {
-        const text = (await option.getText()).trim();
-        allText.push(text);
-        if (text === key) {
-          found = true;
-          await option.scrollIntoView();
-          await option.waitForDisplayed();
-          await waitForElementAnimations(browser, option);
-          await option.click();
-          break;
+      let allText: string[] = [];
+
+      // for whatever reasons sometimes the first one or two come through as empty strings
```

Comment:
I eventually gave up on this and just added retry. My suspicion is that leafygreen doesn't synchronously render all the select options' text because all the options are there but some have blank text. I thought it was maybe some kind of virtual scrolling at first, but all the options are there (just some without text) and they do all appear by themselves on the retry. Almost like a weird optimisation where something renders just the ones that are visible first and then the others later? That sounds like crazytalk :)